lib/macho/load_commands.rb

```diff
@@ -399,6 +339,18 @@ def initialize(view, cmd, cmdsize, name, timestamp, current_version, compatibili
       @current_version = current_version
       @compatibility_version = compatibility_version
     end
+
+    # @param endianness [Symbol, nil] the endianness of the serialized string,
+    #  or nil to use the view's endianness
+    # @return [String] the serialized fields of the load command
+    # @api private
+    def serialize(endianness = nil)
+      endianness ||= view.endianness
```

Comment:
One problem here is that `endianness` alone isn't sufficient to get the proper serialization. To me it doesn't feel that inelegant to tell the `serialize` method something like “here's the file that you will be serialized into; please query this file for its characteristics”. But I don't mind adding an intermediate class, let's call it `SerializationContext`, that gets populated by the `MachOFile` with its endianess and bit width properties and then that instance of `SerializationContext` is passed to the load command's `serialize` method. How does that sound?